src/sentry/runner/commands/devservices.py

```diff
@@ -105,24 +99,19 @@ def up(project, exclude):
             )
             exclude |= {"kafka", "zookeeper", "snuba", "clickhouse"}
 
-    if not settings.SENTRY_USE_RELAY:
-        click.secho(
-            "! Skipping relay, and reverse_proxy since you are not using Relay.",
-            err=True,
-            fg="cyan",
-        )
-        exclude |= {"relay", "reverse_proxy"}
-
-    if not sentry_options.get("symbolicator.enabled"):
-        exclude |= {"symbolicator"}
-
     get_or_create(client, "network", project)
 
     containers = {}
     for name, options in settings.SENTRY_DEVSERVICES.items():
         if name in exclude:
             continue
         options = options.copy()
+
+        test_fn = options.pop("only_if", None)
+        if test_fn and not test_fn(settings, sentry_options):
```

Comment:
nit: `if test_fn is not None`

FWIW I generally don't like truthiness and in my experience, it's caused huge problems with the py3.

Reply:
Or I guess you could just directly call `test_fn` because a runtime error seems fine in this context to me. Maybe add a friendly message like "you've misconfigured `SENTRY_DEVSERVICES`"

Reply:
nit: use truthiness instead of type checks unless it makes absolute sense :)

aside

```
>>> timeit.timeit('for _ in xrange(10000): not None', number=10000)
1.5487449169158936
>>> timeit.timeit('for _ in xrange(10000): None is not None', number=10000)
1.6873209476470947
```

Reply:
> aside
> 
> ```
> >>> timeit.timeit('for _ in xrange(10000): not None', number=10000)
> 1.5487449169158936
> >>> timeit.timeit('for _ in xrange(10000): None is not None', number=10000)
> 1.6873209476470947
> ```

lol wtf i'd have thought pointer comparison would have been obviously faster than truthiness